### PR TITLE
domain: Add reload timeout

### PR DESCRIPTION
**Pull request by zimulala**

### CODE REVIEW

File: domain/domain.go [hunk 1]
```diff
@@ -142,20 +138,26 @@ func (do *Domain) tryReload() {
 	}
 }
 
-func (do *Domain) reload() error {
+func (do *Domain) reload(retCh chan error) error {
 	err := kv.RunInNewTxn(do.store, false, do.loadInfoSchema)
```
Comment (siddontang) [inline domain/domain.go]: do the timeout check in reload, not outer. 
not only ticker but also others can call reload too.

Comment (zimulala) [inline domain/domain.go]: Gotcha.

File: domain/domain.go [hunk 2]
```diff
@@ -172,22 +174,26 @@ func (do *Domain) loadSchemaInLoop(lease time.Duration) {
...
-			log.Warnf("reload, lease:%v", lease)
-			err := do.reload()
-			// we may close store in test, but the domain load schema loop is still checking,
-			// so we can't panic for ErrDBClosed and just return here.
-			if terror.ErrorEqual(err, localstore.ErrDBClosed) {
-				log.Warnf("reload, lease:%v, err:%v", lease, err)
-				return
-			} else if err != nil {
-				log.Fatalf("reload schema err %v", err)
+			go do.reload(reloadErrCh)
```
Comment (siddontang) [inline domain/domain.go]:
```
done := make(err error, 1)
go func() {
    err := do.reload()
}()

select {
    case err := <-done:  
   case timeout:
}
```

**Comment by siddontang** [comment]
LGTM

**Comment by zimulala** [comment]
PTAL@qiuyesuifeng

**Comment by qiuyesuifeng** [comment]
LGTM

**Comment by zimulala** [comment]
ok

**Pull request merged**